src/app/components/cart/components/cart-modal/cart-modal.component.html

```diff
@@ -0,0 +1,170 @@
+                      class="text-blue-500 hover:text-blue-600 cursor-pointer"
+                      (click)="goToCollection(item.collection.uid)"
+                    >
+                      {{ item.collection.name }}
+                    </span>
+                  </div>
+                  <div>
+                    NFT:
+                    <span
+                      class="text-blue-500 hover:text-blue-600 cursor-pointer"
+                      (click)="goToNft(item.nft.uid)"
+                    >
+                      {{ item.nft.name }}
+                    </span>
+                  </div>
+                  <div>Royalties: {{ (item.collection.royaltiesFee || 0) * 100 }}%</div>
+                </div>
+              </div>
+
+              <div class="text-lg font-bold truncate flex-1 mr-4">
+                <div class="text-xs font-medium text-foregrounds-secondary underline mb-2" i18n>
+                  Status
+                </div>
+                {{ cartItemsStatus[i] }}
+              </div>
+
+              <div class="text-lg font-bold truncate flex-1 mr-4">
+                <div class="text-xs font-medium text-foregrounds-secondary underline mb-2" i18n>
+                  Qty Added / Available
+                </div>
+                <div class="flex items-baseline">
+                  <nz-form-item class="grow">
+                    <nz-form-control
+                      i18n-nzErrorTip
+                      nzErrorTip="Minimum 1 and maximum {{ cartItemsQuantities[i] }}."
+                    >
+                      <nz-input-group nzSize="default" [nzSuffix]="suffixMaxQty" class="w-30">
+                        <input
+                          nz-input
+                          type="number"
+                          class="w-full"
+                          nzSize="default"
+                          i18n
+                          [ngModel]="item.quantity"
+                          (change)="updateQuantity($event, item.nft.uid)"
+                          min="0"
+                          max="{{ cartItemsQuantities[i] }}"
+                        />
+                        <ng-template #suffixMaxQty>
+                          <span>/ {{ cartItemsQuantities[i] }}</span>
+                        </ng-template>
+                      </nz-input-group>
+                    </nz-form-control>
+                  </nz-form-item>
+                </div>
+              </div>
+
+              <div class="text-lg font-bold truncate flex-1 mr-4">
+                <div class="text-xs font-medium text-foregrounds-secondary underline mb-2" i18n>
+                  Price Each
+                </div>
+                <ng-container *ngIf="cartItemPrices[item.nft.uid]">
+                  <div
+                    class="mr-2 text-xs font-medium line-through truncate text-foregrounds-secondary"
+                    *ngIf="discount(item.collection, item.nft) < 1"
+                  >
+                    {{
```

Comment:
might be easier to understand if it's a function.

Reply:
Understood, will make sure to keep as much logic in component as possible in future.  

I see elsewhere that formatToken pipe was handled in HTML directly, should I continue that or import the pipe in component and perform the formats there instead?

Reply:
To be honest, some things are not very clean it's just continuum of bad legacy. You can be quite liberal on improving things and coming up with cleaner concepts. Even formatToken might need some improvements. 

My understanding is that it's ideal to use formatToken in the templates instead of components but I might be wrong here.